setuptools/build_meta.py

```diff
@@ -101,7 +101,12 @@ def _file_with_extension(directory, extension):
         f for f in os.listdir(directory)
         if f.endswith(extension)
     )
-    file, = matching
+    try:
+        file, = matching
+    except ValueError:
+        raise ValueError(
+            'No distribution was found. Ensure that `setup.py` '
+            'is not empty and that it calls `setup()`.')
```

Comment:
Should this be `raise ValueError(...) from None`, to simplify the output? The original ValueError from unpacking the tuple is not interesting to anyone and adding noise, since it's literally the only thing done here?

Reply:
That's a good idea.

Reply:
Then again, somehing else might raise a `ValueError` since the generator isn't consumed before it's unpacked and `ValueError` is so ubiquitous that I'm a little wary of erasing it.

Reply:
If you'd like to convert the generator to a list comprehension and then raise from `None`, I think that'd be fine.

Reply:
Since the generator is created on the line above, there currently should be no way to for it to be exhausted before it hits this point?

Reply:
Yes, hence it might raise an unrelated value error in the try block.

Reply:
Ahh, I see what you mean.